library/core/src/array/mod.rs

```diff
@@ -427,3 +427,78 @@ impl<T, const N: usize> [T; N] {
         unsafe { crate::mem::transmute_copy::<_, [U; N]>(&dst) }
     }
 }
+
+/// Creates an array `[T; N]` where each array element `T` is returned by the `cb` call.
+///
+/// # Arguments
+///
+/// * `cb`: Callback where the passed argument is the current array index.
+///
+/// # Example
+///
+/// ```rust
+/// let array: [usize; 5] = core::array::create_array(|i| i);
+/// assert_eq!(array, [0, 1, 2, 3, 4]);
+/// ```
+#[inline]
+#[unstable(feature = "create_array", issue = "0")]
+pub fn create_array<F, T, const N: usize>(mut cb: F) -> [T; N]
+where
+  F: FnMut(usize) -> T,
+{
+    let mut array: MaybeUninit<[T; N]> = MaybeUninit::uninit();
+    // SAFETY: It is guaranteed that the whole array will be filled
+    unsafe {
+        for (idx, value_ptr) in (&mut *array.as_mut_ptr()).iter_mut().enumerate() {
+            ptr::write(value_ptr, cb(idx));
```

Comment:
If `cb` panics, all of the previously created elements will be leaked.

Reply:
Didn't predict this scenario. Thanks for pointing out this problem

Reply:
The functions are using shared code that is already guarded against possible panics. Marking as resolved.